### DISCUSSION TITLE
Client status doesn't update when connecting the wallet.

### DISCUSSION THREAD

**Discussion by Galmoli** [wevm/wagmi#493] [Q&A] [answered]
I am using wagmi to integrate web3 functionalities in a game developed with Unity (hence the `unityContext` events).

I'm trying to get client's connection status with `wagmiClient.status`

```js
import { useEffect } from 'react';
import { unityContext } from '../UnityGame';
import { wagmiClient } from '../../walletConfig';

export const  WalletConnection = () => {
    useEffect(() => {
        unityContext.on("GetWalletConnectionStatus", () => {
            unityContext.send("ReactReceiver" , "SetWalletConnectionStatus", wagmiClient.status);
            console.log(wagmiClient.status);
        });
        return () => {
            unityContext.removeAllEventListeners();
          };
      }, [wagmiClient]);

  return null;
}
```

The code above works as intended when disconnecting the wallet, going from the `connected` status to: `disconnected`. It logs the correct status. But when the current status is `disconnected` and I connect the wallet, it keeps loging `dissconnected`.

To bring some context, I'm using [RainbowKit](https://www.rainbowkit.com/) to connect the wallet. Also `'../../walletConfig'` is just a file to build some objects and export them.

```js
 export {
        WagmiProvider,
        RainbowKitProvider,
        chains,
        wagmiClient
    }
```

**Comment by Galmoli** [comment, marked as answer]
Just found the solution. I was using `wagmiClient` to fetch the connection status. 

The solution is to use the `useConnect` hook.

```js
import { useEffect } from 'react';
import { unityContext } from '../UnityGame';
import { useConnect } from 'wagmi';

export const  WalletConnection = () => {
    var connect = useConnect();
    useEffect(() => {
        unityContext.on("GetWalletConnectionStatus", () => {
            unityContext.send("ReactReceiver" , "SetWalletConnectionStatus", connect.status);
            console.log(connect.status);
        });
        return () => {
            unityContext.removeEventListener("GetWalletConnectionStatus");
          };
      }, [connect]);

  return null;
}
```

**Reply by ted-palmer** [reply]
Do you have a link to this repo? Would love to take a look at this
